packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1316,18 +1339,19 @@ class _DateRangePickerDialogState extends State<DateRangePickerDialog> with Rest
 
   @override
   Widget build(BuildContext context) {
+    final bool useMaterial3 = Theme.of(context).useMaterial3;
```

Comment:
Add a `final ThemeData theme = Theme.of(context);` and use it here, and below where the `dialogTheme` is fetched.

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/70d860d720f506a57c730749eaaba252e4cd8a6b